andi/errors.py

```diff
@@ -1,4 +1,66 @@
 # -*- coding: utf-8 -*-
+from collections import namedtuple
+
+
 class NonProvidableError(TypeError):
     """ Raised when a type is not providable """
 
+    def __init__(self, class_or_func, errors_per_argument):
+        self.class_or_func = class_or_func
+        self.errors_per_argument = errors_per_argument
+        super().__init__(_exception_msg(class_or_func, errors_per_argument))
+
+
+CyclicDependencyErrCase = namedtuple("CyclicDependencyErrCase",
```

Comment:
@kmike should I declare these as private? I'm not entirely clear regarding when to make public or private.